content/blog/is-dns-migration-a-tricky-affair.md

```diff
@@ -0,0 +1,106 @@
+---
+
+title: "Is DNS migration a tricky affair?"
+
+date: 2022-12-15
+
+draft: false
+
+featured: true
+
+weight: 1
+
+---
+
+Never trust a friend who says DNS migration is an easy affair. Your instinct says it's easy, well don't trust your instinct. In this post, I will be sharing my experience with one such encounter with DNS migration from GoDaddy to Route53.
+
+Before starting, if you are here because of the catchy title and have no idea of what DNS is, or you are one of my many (zero) followers or need to brush up your knowledge on DNS, please refer to these resources:
+1. [What is DNS - Cloudflare](https://www.cloudflare.com/learning/dns/what-is-dns/)
+2. [Deep Dive on DNS - IETF](https://www.youtube.com/watch?v=DV0q9s94RL8)
+
+I won't keep you waiting any longer - I will start my story and explain the reason for my introductory lines.
+# Need for DNS migration
+One of our clients wanted us to move their workload from docker swarm to Amazon ECS. Earlier, they were using HaProxy to load-balance the traffic. This setup also included generation and installation of SSL certificates using in-house scripts, which the client needed to maintain. So, we decided to go with a managed solution to reduce the management overhead and chose AWS ALB and ACM combination. In order for ACM to auto-renew expiring SSL certificates, we decided to use Route53 as the DNS service, ditching GoDaddy's DNS service. BTW, if you are curious, the domain was registered with GoDaddy and hence their DNS service.
+
+These decisions landed us onto the supposedly **_EASY_** task of DNS migration. This was my first DNS migration task, and I was equally afraid as I was excited. Just then, I overheard someone (friend/colleague) saying with a very convincing tone that DNS migration is an easy task. Well, I was somewhat convinced. 😅
+
+Still, as a good engineer 🤭, I went ahead and did my research regarding DNS migration and carved out a migration plan.  Looking at the plan for migration, even my instincts were convinced. 
+
+
+![Instinct calling](/images/blog/is-dns-migration-a-ticky-affair/do_it.gif)
+# Migration Plan
+Basically I divided the migration plan in three phases along with a rollback strategy:
+## Pre-execution phase
+In this phase we exported the [zone file](https://en.wikipedia.org/wiki/Zone_file) from GoDaddy DNS service which we will be referring to for the list of domain records to migrate.
+
+Another thing we did was to reduce the [TTL](https://www.cloudflare.com/learning/cdn/glossary/time-to-live-ttl/) of all the records in GoDaddy DNS to the least possible value. These tasks we performed a day before the execution phase. For us, the maximum value of TTL was one day, so 24 hours was enough for us to reflect new TTL values.
```

Comment:
Maybe rephrase the last line. Sounds little weird, tbh.

Reply:
Fixed in 15acecd